spec/haskell/src/SEL4/Object/Reply.lhs

```diff
@@ -34,51 +34,31 @@ This module specifies the behavior of reply objects.
 >     reply <- getReply replyPtr
 >     setReply replyPtr (upd reply)
 
+> bindScReply :: PPtr SchedContext -> PPtr Reply -> Kernel ()
+> bindScReply scPtr replyPtr = do
+>     sc <- getSchedContext scPtr
+>     scReplyOpt <- return $ scReply sc
+>     when (scReplyOpt /= Nothing) $ do
+>         scReplyPtr <- return $ fromJust scReplyOpt
+>         updateReply scReplyPtr (\reply -> reply { replyNext = Just (Next replyPtr) })
+>     updateReply replyPtr (\reply -> reply { replyPrev = scReplyOpt })
+>     setSchedContext scPtr (sc { scReply = Just replyPtr })
+>     updateReply replyPtr (\reply -> reply { replyNext = Just (Head $ scPtr) })
```

Comment:
My plan with this was to just sort out the differences in CRefine. I haven't worked in CRefine, but I was under the impression that slight reorderings were not too hard to accomodate because the proofs are close to symbolic executions anyway.

Reply:
https://sel4.atlassian.net/browse/VER-1380